### PR TITLE
Discover CD++ simulator executables

### PR DESCRIPTION
**Pull request by plbalbi**
As first step towards de CD++ simulator wrapper, the executable has to be discovered. A compound discoverer is presented, in order to do the lookup of the executable to run. In some future, it event can be easily extended to perform the lookup in some other manner.

### CODE REVIEW

File: colonel/wrapper/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        executable_route = os.path.join(Wrapper.CDPP_LIBRARY_DEFINED_EXECUTABLE_PATH, Wrapper.CDPP_BIN)
+        if ExecutableDiscoverer.is_executable_file(executable_route):
+            return executable_route
+        else:
+            return None
+
+class Wrapper:
+    CDPP_BIN = 'cd++'
+    CDPP_EXECUTABLE_ENV_VAR = 'CDPP_BIN'
+    CDPP_LIBRARY_DEFINED_EXECUTABLE_PATH = '~/Facultad/pringles/cdpp_kernel/bin/'
```
Comment (juanlanuzag) [inline colonel/wrapper/wrapper.py]: Hardcoded! should be param

File: colonel/wrapper/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# sample drawlog command
+# drawlog -m2Voronoi.ma -i00:00:00:100 -cvoronoiExpansion -llogs -z/tmp/altoPerro.npz
+
+class SimulationNotExectutedException(Exception):
+    pass
+
+class SimulationProcessFailedException(CalledProcessError):
+    pass
+
+class SimulationExecutedButFailedException(Exception):
```
Comment (juanlanuzag) [inline colonel/wrapper/wrapper.py]: these exceptions should go in the errors.py file

File: colonel/wrapper/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        if not self.simulationWasExecuted():
+            raise SimulationNotExectutedException()
+        return self.logsFileName
+
+    def getOutputPath(self):
+        if not self.simulationWasExecuted():
+            raise SimulationNotExectutedException()
+        return self.outputFileName
+
+    def simulationWasExecuted(self):
```
[the remark this answers is not in the document]
Comment (juanlanuzag) [reply colonel/wrapper/wrapper.py]: all the getters above should also be properties

File: colonel/wrapper/wrapper.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+class SimulationProcessFailedException(CalledProcessError):
+    pass
+
+class SimulationExecutedButFailedException(Exception):
+    pass
+
+# Common interface
+class ExecutableDiscoverer():
+    def __init__(self):
+        pass
```
Comment (juanlanuzag) [inline colonel/wrapper/wrapper.py]: You don't need to define it if its empty

File: colonel/wrapper/file_helpers.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import time
+
+DEFAULT_TEMP_PATH = '/tmp/'
```
[the remark this answers is not in the document]
Comment (plbalbi) [reply colonel/wrapper/file_helpers.py]: Of course. This came from the migrated part of the wrapper. Will clean everything up in next PR.

File: colonel/wrapper/wrapper.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
+import subprocess
```
[the remark this answers is not in the document]
Comment (plbalbi) [reply colonel/wrapper/wrapper.py]: Totally agree. A **requirements.txt** and a **minor-python-version** are needed. Will convert this into an issue.

File: colonel/wrapper/wrapper.py (hunk 1 above)
Comment (plbalbi) [reply colonel/wrapper/wrapper.py]: Moving to issue.

**Pull request merged**